src/AsmResolver.DotNet/TypeDefinition.cs

```diff
@@ -42,6 +42,81 @@ public class TypeDefinition :
+
+        /// <summary>
+        /// Initialize New Delegate.
+        /// </summary>
+        /// <param name="ns">Delegate Namespace.</param>
+        /// <param name="name">Delegate Name.</param>
+        /// <param name="module">Delegate Module.</param>
+        /// <param name="invokeSignature">Delegate.Invoke Signature.</param>
+        /// <param name="attributes">Accessibility Attributes.</param>
+        /// <returns>Fully Implemented Delegate.</returns>
+        public static TypeDefinition CreateDelegate(string ns, string name, ModuleDefinition module, MethodSignature invokeSignature, TypeAttributes attributes = TypeAttributes.NotPublic)
+        {
+
+            var importer = new ReferenceImporter(module);
+
+            var asyncResult = importer.ImportType(new TypeReference(module.CorLibTypeFactory.CorLibScope, nameof(System), nameof(IAsyncResult))).ToTypeSignature();
+            var asyncCallback = importer.ImportType(new TypeReference(module.CorLibTypeFactory.CorLibScope, nameof(System), nameof(AsyncCallback))).ToTypeSignature();
+
+            // TODO: Handles Generics.
+            var delegateType = new TypeDefinition(
+                ns,
+                name,
+                TypeAttributes.Sealed | attributes,
+                importer.ImportType(new TypeReference(module.CorLibTypeFactory.CorLibScope, nameof(System), nameof(MulticastDelegate))));
+
+            delegateType.IsAutoLayout = true;
+
+            // Delegate ctor.
+
+            var ctor = new MethodDefinition(".ctor",
+                MethodAttributes.HideBySig | MethodAttributes.SpecialName | MethodAttributes.RuntimeSpecialName | MethodAttributes.ReuseSlot | MethodAttributes.Public,
+                MethodSignature.CreateInstance(module.CorLibTypeFactory.Void, module.CorLibTypeFactory.Object, module.CorLibTypeFactory.IntPtr));
+
+            ctor.IsRuntime = true;
+            delegateType.Methods.Add(ctor);
+
+            var beginInvoke = new MethodDefinition("BeginInvoke",
+                MethodAttributes.HideBySig | MethodAttributes.Virtual | MethodAttributes.NewSlot,
+                MethodSignature.CreateInstance(asyncResult, invokeSignature.ParameterTypes));
+
+            beginInvoke.Signature.ParameterTypes.Add(asyncCallback);
+            beginInvoke.Signature.ParameterTypes.Add(module.CorLibTypeFactory.Object);
+
+            beginInvoke.IsRuntime = true;
+            delegateType.Methods.Add(beginInvoke);
+
+            var endInvoke = new MethodDefinition("EndInvoke",
+                MethodAttributes.HideBySig | MethodAttributes.Virtual | MethodAttributes.NewSlot,
+                MethodSignature.CreateInstance(module.CorLibTypeFactory.Void, asyncResult));
+
+            endInvoke.IsRuntime = true;
+            delegateType.Methods.Add(endInvoke);
+
+            var invoke = new MethodDefinition("Invoke",
+                MethodAttributes.HideBySig | MethodAttributes.Virtual | MethodAttributes.NewSlot,
+                invokeSignature);
```

Comment:
This assumes `invokeSignature.HasThis` is `true`. This is not the case if we pass on a signature that is e.g. created using `MethodSignature.CreateStatic`. Maybe a check for this and throw an `ArgumentException` if not `true`? A unit test is probably also a good idea for this.